src/ica.jl

```diff
@@ -60,7 +60,7 @@ function fastica!(W::DenseMatrix{Float64},      # initialized component matrix,
                   X::DenseMatrix{Float64},      # (whitened) observation sample matrix, size(m, n)
                   fun::ICAGDeriv,               # approximate neg-entropy functor
                   maxiter::Int,                 # maximum number of iterations
-                  tol::Real,                    # convergence tolerance
+                  tol::Float64,                    # convergence tolerance
```

Comment:
Better not restrict accepted types.

Reply:
This was a workaround because `ConvergenceException` expects `tol` & `chg` to be the same type but can do a conversion instead.

Reply:
It does not matter. `fastica!` works only with `Float64` data - see above types of `W` and `X` parameters. `ica` should be refactored to accept data of any floating type.

Reply:
Yeah, but `tol` could be of a different type from the data itself, as long as comparison operations are supported.